update.go

```diff
@@ -216,7 +215,6 @@ other options are ignored.
 				opt  string
 				dest *uint64
 			}{
-				{"cpu-burst", r.CPU.Burst},
 				{"cpu-period", r.CPU.Period},
 				{"cpu-rt-period", r.CPU.RealtimePeriod},
 				{"cpu-share", r.CPU.Shares},
```

Comment:
Shouldn't we distinguish nil from zero for all the fields?

See https://github.com/opencontainers/runc/pull/4142/commits/c9ba576e41cb39a8e82f0b5dc6155ea718703a6c

Reply:
Yes, I think we should deal with all this type field. Could you please provide your PR link?

Reply:
- https://github.com/opencontainers/runc/pull/4142

Reply:
ping @lifubang @kolyshkin PTAL

Reply:
cherry-picked from your PR and add a testcase.